# Arène des Algos

Projet réalisé dans le cadre du cours de Machine Learning.

L'objectif est de comparer plusieurs algorithmes sur différents problèmes afin d'observer leurs performances et leurs comportements.

## Phase 1 : Découverte du dataset

Avant de construire le moindre modèle, il est utile de comprendre les données avec lesquelles nous allons travailler.

Nous allons utiliser le dataset Breast Cancer fourni par scikit-learn et observer sa structure générale ainsi que la répartition des classes.

In [27]:
from sklearn.datasets import load_breast_cancer
import pandas as pd

In [28]:
def explorer_dataset():

    X, y = load_breast_cancer(return_X_y=True)

    print(f"Lignes, colonnes : {X.shape}")

    repartition = pd.Series(y).value_counts().sort_index()

    for classe, nombre in repartition.items():

        if classe == 0:
            print(f"Classe {classe} (maligne) : {nombre} cas")
        else:
            print(f"Classe {classe} (bénigne) : {nombre} cas")

In [29]:
explorer_dataset()

Lignes, colonnes : (569, 30)
Classe 0 (maligne) : 212 cas
Classe 1 (bénigne) : 357 cas


### Répartition en pourcentage

L'effectif brut permet d'observer le nombre de cas présents dans chaque catégorie.

Une vue en pourcentage permet cependant de mieux évaluer l'équilibre global du dataset.

In [30]:
def afficher_repartition_pourcentage(y):

    repartition = (
        pd.Series(y)
        .value_counts(normalize=True)
        .sort_index()
        * 100
    )

    for classe, pourcentage in repartition.items():
        print(f"Classe {classe} : {pourcentage:.2f}%")

In [31]:
X, y = load_breast_cancer(return_X_y=True)

afficher_repartition_pourcentage(y)

Classe 0 : 37.26%
Classe 1 : 62.74%


### Observations

Le dataset contient 569 observations réparties en deux classes.

Les tumeurs bénignes sont légèrement plus nombreuses que les tumeurs malignes, mais l'écart reste raisonnable.

Ce jeu de données constitue donc un bon candidat pour les premières expérimentations de classification supervisée.